storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/ConstraintSolverStrategy.java

```diff
@@ -331,6 +331,10 @@ protected SolverResult backtrackSearch(SearcherState state) {
             return new SolverResult(state, false);
         }
 
+        if (Thread.currentThread().isInterrupted()) {
```

Comment:
Why do we need to this? (I don't know how this is used and trying to learn)

Reply:
My remembrance from previous googling is that a cancel on the future causes the thread to be interrupted, but it is up to that thread to respond to the interrupt.  If the scheduling thread is stuck in a long loop, it will just keep running.

Reply:
Thanks!